src/storage/mod.rs

```diff
@@ -14,6 +14,7 @@ use std::io::Read;
 use std::path::{Path, PathBuf};
 
 const MAX_CONCURRENT_UPLOADS: usize = 1000;
```

Comment:
Unrelated, but it occurred to me that we could potentially benefit from dynamically selecting the number of concurrent uploads based on the current conditions, e.g. decrease the number of uploads if we're under heavier load